### PR TITLE
[WIP] Add SwinIR Backbone

### PR DESCRIPTION
**Pull request by nijkah**
Support [SwinIR](https://github.com/JingyunLiang/SwinIR)

- [x] run `tools/train.py` and check that oss decreases
- [ ] reproduce the paper result
- [ ] Fix markdown

### CODE REVIEW

**Comment by nijkah** [comment]
I trained the model given the configuration file.
And I found that it is inferior to the paper result, but outperforms other models in mmedit.
In this case, should I postpone uploading the model until the performance is reproduced?

Result is like this
| Method                     | Set5   | Set14  |
|------------------------|--------|------------|
| Classical SwinIR-x2    |   36.0936/0.9386        |            |
| Classical SwinIR-x4    | 30.6932/ 0.8570        |            |

**Comment by nijkah** [comment]
I found that [original training code](https://github.com/cszn/KAIR/) adopts more diverse augmentations.

**Comment by wangruohui** [comment]
Hello,

We can convert the checkpoint from the official release and try to reproduce test accuracy first. In the meantime, we can implement other augmentation techniques to reproduce training accuracy.

**Comment by wangruohui** [comment]
Hello, 

Are you still working on this feature?

**Comment by nijkah** [comment]
@wangruohui Sorry, I was busy with other tasks these days. I may start to continue this next week.
If I cannot continue this PR, I will close it :(.

**Comment by nijkah** [comment]
@wangruohui I think I cannot work on this for a while. I'm even not sure to finish this.
Now, I'll close this PR and re-open it when I finish the reproduction and other issues.
